src/common/snake-case.ts

```diff
@@ -19,6 +21,18 @@ export function toSnakeCase(
      * to "graphql".
      */
     compoundWords?: string[];
+
+    /**
+     * Whether to convert to singular or plural snake case.
+     */
+    pluralize?: boolean;
+
+    /**
+     * A set of irregular plurals to use when converting
+     * to plural snake case. Both the singular and plural
+     * forms should already be in snake case.
```

Comment:
```suggestion
     * A mapping from singular form to irregular plural form,
     * for use when `pluralize` is true. Both forms should be
     * in snake case.
     * Example: `{ bill_of_lading: "bills_of_lading" }`
```

Reply:
That's a much better irregular plural example that I used in the test case—stealing it!